project-koku/koku
# COST-680: Add tag enablement to GCP.

Testing:


1. Run `make docker-reinitdb-with-sources`
2. Run `make load-test-customer-data start=2021-02-01`
3. Go check out the database and check the `reporting_gcpenabledtagkeys` table and make sure all the tags are there, there should be 5:
 ```
test_storage_key
vm_label_proj2
```
4. Check out the tags on the `reporting_gcpcostentrylineitem_daily_summary` table and make sure you see the appropriate distinct tags for the data.

5. Check the enabled tags & cost endpoints:
- http://localhost:8000/api/cost-management/v1/tags/gcp/?filter[enabled]=true
- http://localhost:8000/api/cost-management/v1/reports/gcp/costs/?filter[tag:test_storage_key]=test_storage_label
Note the total costs for gcp:  `http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/`

6. Delete one of the tags from the `reporting_gcpenabledtagkeys` table.

7. Resummarize the data
- http://127.0.0.1:5000/api/cost-management/v1/report_data/?provider_uuid=*&provider_type=GCP-local&schema=acct10001&start_date=2021-02-01

8. Check the distinct tags from the `reporting_gcpcostentrylineitem_daily_summary` table again and make sure the tags do not include the disabled tag. 
9. Check the aws cost and ensure that they have not changed.
